src/schema.ts

```diff
@@ -261,31 +272,65 @@ UserSchema.virtual("oldEnoughForWithdrawal").get(function (this: typeof UserSche
   return d - this.created_at.getTime() > yamlConfig.limits.oldEnoughForWithdrawal
 })
 
-UserSchema.methods.limitHit = async function ({
-  on_us,
-  amount,
-}: {
-  on_us: boolean
-  amount: number
-}) {
-  const timestampYesterday = Date.now() - MS_PER_DAY
+UserSchema.virtual("twoFactorEnabled").get(function (this: typeof UserSchema) {
+  return !!this.twoFactor.secret
+})
 
-  const txnType = on_us
-    ? [{ type: "on_us" }, { type: "onchain_on_us" }]
-    : [{ type: { $ne: "on_us" } }]
+const getTimestampYesterday = () => Date.now() - MS_PER_DAY
 
-  const limit = yamlConfig.limits[on_us ? "onUs" : "withdrawal"].level[this.level]
+// FIXME: Improve function name
+UserSchema.methods.remainingTwoFactorLimit = async function () {
+  const threshold = this.twoFactor.threshold
+
+  const txnType = [
+    { type: "on_us" },
+    { type: "onchain_on_us" },
+    { type: "onchain_payment" },
+    { type: "payment" },
+  ]
 
   const outgoingSats =
     (
       await User.getVolume({
-        after: timestampYesterday,
+        after: getTimestampYesterday(),
         txnType,
         accounts: this.accountPath,
       })
     )?.outgoingSats ?? 0
```

Comment:
I think the getVolume function could be improved. the fact that we need to go `?.outgoingSats ?? 0` to parse the value of the function is not great. this should probably be integrated within the function itself.

Reply:
I've refactored getVolume to have the check inside the function itself